doc/docs/master/deploy-manage/manage/s3gateway/deploy-s3gateway-sidecar.md

```diff
@@ -0,0 +1,91 @@
+pipeline worker pod, Kubeflow can access the files stored in S3 buckets
+in the pipeline pod, which ensures the provenance is maintained
+correctly. The S3 gateway sidecar instance is created together with the
+pipeline and shut down when the pipeline is destroyed.
+
+## Limitations
+
+Pipelines exposed through a sidecar S3 gateway have the following limitations:
+
+* Because of PFS limitations, you cannot write to the input repos exposed
+through the S3 gateway instance. Similarly, you cannot read files from
+the output bucket that you enabled by using the `s3_out` parameter in the
+pipeline spec. This limitation guarantees that pipeline provenance is
+preserved.
+
+* The `glob` field in the pipeline spec should not be specified. The glob
+pattern is explicitly set to `\` and all files are processed as a single
+datum. In this configuration, already processed datums are not skipped which
+could be a performance consideration for some deployments.
+
+* You can create a single or a cross input. Join and union inputs are not
+supported. However, if you combine an S3-enabled input with a non-S3 input,
+you can specify a glob pattern for the latter.
+
+## Expose a Pipeline through an S3 Gateway in a Sidecar
+
+When you work with platforms like Kubeflow or Apache Spark, you need
+to spin up an S3 gateway instance that runs alongside the pipeline worker
+pod as a sidecar container. To do so, set the `s3` parameter in the `input`
+part of your pipeline specification to `true`. When enabled, this parameter
+mounts S3 buckets for input repositories in the S3 gateway sidecar instance
+instead of in `/pfs/`. You can set this property for each PFS input in
+a pipeline. The address of the input repository will be `s3://<input_repo>`.
+
+You can also expose the output repository through the same S3 gateway
+instance by setting the `s3_out` property to `true` in the `root` of
+the pipeline spec.  If set to `true`, Pachyderm creates another S3 bucket
+on the sidecar, and the output files will be written there instead of
+`/pfs/out`. By default, `s3_out` is set to `false`. The address of the
+output repository will be `s3://<output_repo>`.
+
+You can connect to the S3 gateway sidecar instance through its Kubernetes
+service. To access the sidecar instance and the buckets on it, you need
+to know the address of the buckets. Because PPS handles all permissions,
+no special authentication configuration is needed.
+
+Both the master and sidecar gateways can run concurrently, as well as
+repositories can be accessed through both of them.
+
+The following text is an example of a pipeline exposed through a sidecar
+S3 gateway instance:
+
+```json
+{
+	 "pipeline": {
+	    "name": "test"
+	 },
+	 "input": {
+	    "pfs": {
+	       "glob": "/*",
+	       "repo": "test-repo",
+           "s3": "true"
+	 },
+	 "transform": {
+	    "cmd": [ "python3", "/test.py" ],
+	    "image": "pachyderm/test"
+	 },
+     "s3_out": "true"
+}
```

Comment:
this could use some consistent tabbing

Reply:
fixed

Reply:
I think we should add something like:

For the above example, your code should read data from `s3://images` and should write to `s3://edges`.

Reply:
so, the `"s3_out": `parameter  I think is either true or false. not a path @ysimonson , is this correct?

Reply:
it is true or false, but it's supposed to be a boolean rather than a string, e.g. `"s3_out": true`